Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -99,12 +127,44 @@ public float Distance
         private int rows = 3;
 
         /// <summary>
-        /// Number of rows per column, column number is automatically determined
+        /// Number of rows per column.false Can only be assigned when layout type is
+        /// RowsThenColumns
         /// </summary>
         public int Rows
         {
             get { return rows; }
-            set { rows = value; }
+            set 
+            {
+                if (Layout == LayoutOrder.ColumnThenRow) 
+                {
+                    Debug.LogError("Cannot assign Rows when layout type is " + Layout);
+                    return;                    
+                }
+                rows = value; 
+            }
+        }
+
+        // protected int Columns;
+        [Tooltip("Number of columns per row")]
+        [SerializeField]
+        private int columns = 3;
+
+        /// <summary>
+        /// Number of columns per row. Can only be assigned when layout type is 
+        /// ColumnsThenRows
+        /// </summary>
+        public int Columns
+        {
+            get { return columns; }
+            set 
+            { 
+                if (Layout == LayoutOrder.RowThenColumn)    
+                {
+                    Debug.LogError("Cannot assign Columns when layout type is " + Layout);
```

Comment:
Same comment as above with respect to error message messaging (staying in the positive instead of the negative)

Reply:
Let me know what you think of the  new error messages.